packages/application-generic/src/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts

```diff
@@ -153,12 +157,74 @@ export class GetSubscriberTemplatePreference {
   }
 }
 
-function filterActiveChannels(
+function updateOverrideReasons(
+  channelName,
+  sourceName: string,
+  index: number,
+  overrideReasons: IPreferenceOverride[]
+) {
+  const currOverride: IPreferenceOverride = {
+    channel: channelName as ChannelTypeEnum,
+    source: sourceName as 'template' | 'subscriber',
+  };
+
+  const notFoundFlag = -1;
+  const existsInOverrideReasons = index !== notFoundFlag;
+  if (existsInOverrideReasons) {
+    overrideReasons[index] = currOverride;
+  } else {
+    overrideReasons.push(currOverride);
+  }
+}
+
+export function determineOverrides(
```

Comment:
the main logic of the template preferences. i centralized the logic to cannon for loop so it will be easier (i hope) to understand.

Reply:
for me it was hard to read, but I think that if you will split it to smaller functions with a good naming and will do the other suggestions it will be much better